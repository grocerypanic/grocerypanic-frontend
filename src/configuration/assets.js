const Assets = {
  favicon: `${process.env.PUBLIC_URL}/favicon.ico`,
  nonBreakingSpace: "\u00A0",
  cart: `${process.env.PUBLIC_URL}/images/cart_desktop.jpg`,
  bill: `${process.env.PUBLIC_URL}/images/bill_desktop.jpg`,
  cart_mobile: `${process.env.PUBLIC_URL}/images/cart_mobile.jpg`,
  bill_mobile: `${process.env.PUBLIC_URL}/images/bill_mobile.jpg`,
};

export default Assets;
