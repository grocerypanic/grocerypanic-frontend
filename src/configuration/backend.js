export const SafeMethods = ["GET", "HEAD", "OPTIONS", "TRACE"];
export const ItemFilters = ["preferred_stores", "shelf"];
export const FilterTag = "filter_tag";

export const Constants = {
  csrfLocalStorage: "csrf-token",
  csrfHeaderName: "X-CSRFToken",
  genericAPIError: "API Error",
  csrfErrorMessage: "Refresh csrf and try again.",
  alreadyRegistered: "User is already registered with this e-mail address.",
  minimumQuanity: 0,
  maximumQuantity: 10000,
  minimumShelfLife: 1,
  maximumShelfLife: 365 * 3,
  defaultShelfLife: 7,
  minimumPrice: 0,
  maximumPrice: 9999,
  maximumTransactionHistory: 14,
  AnalyticsCookieName: "PanicDataConsent-" + process.env.NODE_ENV.toUpperCase(),
  duplicateObjectApiError: { non_field_errors: ["Object already exists."] },
  pageLookupParam: "page",
  pageOverrideParam: "all_results",
};

export const Providers = {
  google: "google",
  facebook: "facebook",
};

export const Paths = {
  googleLogin: "/api/v1/auth/social/google/",
  facebookLogin: "/api/v1/auth/social/facebook/",
  manageShelves: "/api/v1/shelves/",
  manageStores: "/api/v1/stores/",
  manageItems: "/api/v1/items/",
  refreshCSRF: "/api/v1/auth/csrf/",
  manageTransactions: "/api/v1/transactions/",
};

const month = 30;
const year = 365;

export const ShelfLifeConstants = [
  { name: "3 Days", id: 3 },
  { name: "1 Week", id: 7 },
  { name: "2 Weeks", id: 14 },
  { name: "3 Weeks", id: 21 },
  { name: "1 Month", id: month },
  { name: "2 Months", id: month * 2 },
  { name: "3 Months", id: month * 3 },
  { name: "6 Months", id: month * 6 },
  { name: "1 Year", id: year },
  { name: "2 Years", id: year * 2 },
  { name: "3 Years", id: year * 3 },
];
