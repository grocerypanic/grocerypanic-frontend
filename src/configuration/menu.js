import Routes from "./routes";

const Options = [
  { name: "All Items", location: Routes.items },
  { name: "Shelves", location: Routes.shelves },
  { name: "Stores", location: Routes.stores },
];

export default Options;
