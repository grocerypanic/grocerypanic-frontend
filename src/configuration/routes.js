const Routes = {
  root: "/",
  menu: "/menu",
  about: "/about",
  privacy: "/privacy",
  signin: "/signin",
  shelves: "/shelves",
  splash: "/",
  stores: "/stores",
  items: "/items",
  create: "/create",
  details: "/details/:id",
  goBack: "back",
};

export const External = {
  feedback: "https://www.sharedvisionsolutions.com/#contact",
  credit: "https://www.sharedvisionsolutions.com/",
};

export default Routes;
