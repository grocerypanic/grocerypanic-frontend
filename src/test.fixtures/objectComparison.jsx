export const propCount = (hash, length) => {
  expect(Object.keys(hash).length).toBe(length);
};
