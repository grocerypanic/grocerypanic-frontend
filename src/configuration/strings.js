import assets from "./assets";

export const Strings = {
  en: {
    SignIn: {
      Title: "Sign In",
      LoginMessageGoogle: "Login With Google",
      LoginMessageFacebook: "Login With Facebook",
      ErrorLoginFailure: "Unable to authenticate!\nPlease try again.",
      ErrorAuthExpired: "Authentication Expired!\nPlease login again.",
      ErrorDuplicateAccount:
        "You have already signed up!\nTry logging in with another social account.",
      PendingSocialConnection: "Connecting . . .",
    },
    ErrorDialogue: {
      ErrorDialogueConfirm: "OK",
      ErrorDialogueTitle: "Error!",
    },
    Copyight: {
      CopyrightDeclaration: "Copyright Â© ",
      CopyrightMessage: "Your Website",
    },
    App: {
      Suspense: "Suspense filled message...",
    },
    CookiePolicy: {
      CookieMessage:
        "This website would like to use cookies to measure how you as an individual interact with it.  We'll use this information to improve the service.",
      CookieAcceptText: "I'm ok with that",
      CookieDeclineText: "I Decline",
    },
    MainMenu: {
      Title: "Main Menu",
      HeaderTitle: "Panic: Main Menu",
      HelpText: "Make a selection to proceed.",
    },
    ItemDetails: {
      Title: "Edit Item Details",
      HeaderTitle: "Details",
      HelpText:
        "Fill out the values, click the save button to finalize.\nDon't modify quantity here.",
      NameLabel: "Name",
      QuantityLabel: "Qty",
      PriceLabel: "Price",
      StoresLabel: "Stores",
      SaveButton: `${assets.nonBreakingSpace}Save${assets.nonBreakingSpace}`,
      DeleteButton: "Delete",
      MultiSelectHelp: "",
      ShelvesDetail: "Where do you keep this at home?",
      ShelfLifeDetail: "How long does this item typically keep?",
      PerferredLocationDetails: "Where do you prefer to buy this?",
      ErrorUnselectedStore: "Choose a preferred stored.",
      SaveAction: "Saved",
      DeleteAction: "Deleting ... :(",
      Tabs: {
        Edit: "Edit",
        Stats: "Stats",
      },
      ApiError:
        "Unable to retrieve data from the API!\nDoes this content really exist?\nRetry, or change your query.",
    },
    ItemList: {
      ApiError:
        "Unable to retrieve data from the API!\nDoes this content really exist?\nRetry, or change your query.",
    },
    InventoryPage: {
      Title: "Inventory (All)",
      HeaderTitle: "Inventory",
      HelpText:
        "Click and hold the item's name for details.\nUse the create button (upper right) to add new items.\nIncrese or decrease quantity with the buttons.",
      Save: "save",
      Delete: "delete",
      PlaceHolderMessage: "You have no items yet.",
      ErrorInsufficientInventory: "You don't have that many.",
      Quantity: {
        Title: "Quantity",
        Message: "This is how many items you have in stock.",
      },
      Expired: {
        Title: "Expired",
        Message:
          "This is how many items you have in stock that may be expired.",
      },
    },
    SimpleList: {
      ValidationAlreadyExists: "This already exists",
      ValidationFailure: "Enter a valid name",
      SaveButton: "save",
      DeleteButton: "delete",
      CreatedAction: "Created",
      DeletedAction: "Deleted",
    },
    ShelfPage: {
      Title: "Your Shelves",
      HeaderTitle: "Shelves",
      PlaceHolderMessage: "You Have No Shelves Yet.",
      HelpText:
        "Click a shelf to select, click and hold to delete.\nUse the create button (upper right) to add new shelves.",
    },
    StorePage: {
      Title: "Your Stores",
      HeaderTitle: "Stores",
      PlaceHolderMessage: "You Have No Stores Yet.",
      HelpText:
        "Click a store to select, click and hold to delete.\nUse the create button (upper right) to add new stores.",
    },
    Testing: {
      GenericTranslationTestString: "GenericTranslationTestString",
      GenericMultiLineTranslationTestString:
        "GenericTranslationTestString\nGenericTranslationTestString",
    },
    PlaceHolder: {
      PlaceHolderMessage: "I'm only a placeholder.",
    },
  },
};

// Export english as default for matching during tests
export default Strings.en;
