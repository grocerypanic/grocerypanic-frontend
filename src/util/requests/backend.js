import { Constants, SafeMethods } from "../../configuration/backend";
import debug from "../debug";

const Backend = (method, path, data = null) => {
  // Prepend Backend Hostname and Protocol If Needed
  let url;
  if (!path.includes(process.env.REACT_APP_PANIC_BACKEND)) {
    url = `${process.env.REACT_APP_PANIC_BACKEND}${path}`;
  } else {
    url = path;
  }

  debug(`API ${method}:\n ${url}`);
  if (data) debug(`Body:\n ${JSON.stringify(data)}`);
  let csrf;
  let statusCode;

  const options = {
    method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    credentials: "include",
  };

  if (data) options.body = JSON.stringify(data);
  if (!SafeMethods.includes(method)) {
    csrf = localStorage.getItem(Constants.csrfLocalStorage);
    if (csrf) options.headers[Constants.csrfHeaderName] = csrf;
  }

  const response = fetch(url, options)
    .then(async (fetchResponse) => {
      const contentType = fetchResponse.headers.get("content-type");
      statusCode = fetchResponse.status;
      if (contentType === null) return null;
      if (contentType.startsWith("application/json"))
        return await fetchResponse.json();
      return fetchResponse.text();
    })
    .then((fetchResponseDecoded) => {
      debug(`API Response Status Code:\n ${statusCode}`);
      if (fetchResponseDecoded) {
        debug(`API Response Data:\n`);
        debug(fetchResponseDecoded);
      }
      return [fetchResponseDecoded, statusCode];
    })
    .catch((err) => {
      return [Constants.genericAPIError, 500];
    });
  return response;
};

export default Backend;
