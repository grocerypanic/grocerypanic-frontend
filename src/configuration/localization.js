import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { Strings } from "./strings";

const i18nConfig = {
  fallbackLng: {
    default: ["en"],
  },
  debug: true,
  ns: ["translation"],
  defaultNS: "translation",
  keySeparator: ".",
  interpolation: {
    escapeValue: false,
    formatSeparator: ",",
  },
  react: {
    wait: true,
  },
};

i18n.use(LanguageDetector).init({
  resources: {
    ...Strings,
  },
  ...i18nConfig,
});

export default i18n;
