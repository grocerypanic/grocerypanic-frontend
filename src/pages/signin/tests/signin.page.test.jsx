import React from "react";
import { render, waitFor, cleanup } from "@testing-library/react";
import { propCount } from "../../../test.fixtures/objectComparison";

import * as Icons from "@material-ui/icons/LockOutlined";
import {
  GoogleLoginButton,
  FacebookLoginButton,
} from "react-social-login-buttons";

import SocialLoginController from "../../../components/social-login-controller/social-login-controller.container";
import Copyright from "../../../components/copyright/copyright.component";
import Signin from "../signin.page";

import Strings from "../../../configuration/strings";
import { Providers } from "../../../configuration/backend";

jest.mock(
  "../../../components/social-login-controller/social-login-controller.container",
  () => ({
    __esModule: true,
    default: jest.fn(),
    providers: jest.requireActual(
      "../../../components/social-login-controller/social-login-controller.container"
    ).providers,
  })
);
jest.mock("@material-ui/icons/LockOutlined", () => ({
  __esModule: true,
  default: jest.fn(),
}));
jest.mock("../../../components/copyright/copyright.component");
const handleSocialLogin = jest.fn();
const handleSocialLoginError = jest.fn();

Icons.default.mockImplementation(() => <div>MockLock</div>);
SocialLoginController.mockImplementation(() => <div>SocialLoginButton</div>);
Copyright.mockImplementation(() => <div>MockCopyRight</div>);

const originalEnvironment = process.env;

describe("Setup Environment", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    process.env.REACT_APP_FACEBOOK_ACCOUNT_ID = "MOCK_FACEBOOK_ACCOUNT";
    process.env.REACT_APP_GOOGLE_ACCOUNT_ID = "MOCK_GOOGLE_ACCOUNT";
    render(
      <Signin
        handleSocialLogin={handleSocialLogin}
        handleSocialLoginError={handleSocialLoginError}
      />
    );
  });
  afterAll(() => {
    process.env = originalEnvironment;
  });

  afterEach(cleanup);

  it("should render all the subcomponents as expected", async (done) => {
    await waitFor(() => expect(Icons.default).toBeCalledTimes(1));
    expect(SocialLoginController).toBeCalledTimes(2);
    expect(Copyright).toBeCalledTimes(1);
    done();
  });

  it("should render the google social login component with correct props", async (done) => {
    await waitFor(() => expect(SocialLoginController).toBeCalledTimes(2));

    // Google Button
    const call1 = SocialLoginController.mock.calls[0][0];
    propCount(call1, 6);
    expect(call1.ButtonType).toBe(GoogleLoginButton);
    expect(call1.appId).toBe(process.env.REACT_APP_GOOGLE_ACCOUNT_ID);
    expect(call1.message).toBe(Strings.LoginMessageGoogle);
    expect(call1.provider).toBe(Providers.google);
    expect(call1.onLoginSuccess).toBe(handleSocialLogin);
    expect(call1.onLoginFailure).toBe(handleSocialLoginError);
    done();
  });

  it("should render the facebook social login component with correct props", async (done) => {
    await waitFor(() => expect(SocialLoginController).toBeCalledTimes(2));

    // Facebook Button
    const call2 = SocialLoginController.mock.calls[1][0];
    propCount(call2, 6);
    expect(call2.ButtonType).toBe(FacebookLoginButton);
    expect(call2.appId).toBe(process.env.REACT_APP_FACEBOOK_ACCOUNT_ID);
    expect(call2.message).toBe(Strings.LoginMessageFacebook);
    expect(call2.provider).toBe(Providers.facebook);
    expect(call2.onLoginSuccess).toBe(handleSocialLogin);
    expect(call2.onLoginFailure).toBe(handleSocialLoginError);
    done();
  });
});
