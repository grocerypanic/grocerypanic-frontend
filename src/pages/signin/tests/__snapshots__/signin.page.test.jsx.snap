// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment should match the snapshot on file (styles) 1`] = `
.c0 {
  margin-top: calc(50px + 26px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c3 {
  padding-bottom: 10px;
  min-width: 250px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #006600;
  color: #FFFFFF;
  text-align: center;
  overflow: hidden;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div>
          MockLock
        </div>
      </div>
      <h1>
        Sign In
      </h1>
      <div
        class="c3"
      >
        <div>
          SocialLoginButton
        </div>
      </div>
      <div
        class="c3"
      >
        <div>
          SocialLoginButton
        </div>
      </div>
      <div>
        MockCopyRight
      </div>
    </div>
  </div>
</div>
`;
